rfcs/047-nested-columns.md

```diff
@@ -0,0 +1,189 @@
+
+You can expand/collapse all nesting using the text button at top right corner.
+
+Please note that ⌄ icon is now used for expanding each level of nesting, rather than opening the column details within the list. Instead, when you click on the name of any column, you will see all its metadata in the dedicated right side panel. You can always close the panel to see more of the list.
+
+The hierarchy of nested objects without named fields (e.g. Array and Map) are now better specified and distinguished from nested objects with named fields (e.g. Struct).
+
+
+## UI/UX-level Explanation
+
+### Expand/collapse all + each level drill down
+
+![Default all expanded view](../assets/047/default-all-expanded-view.png)
+
+* As default, all nested columns will be expanded, showing the option for user to ‘Collapse all nested’
+* For most complex tables, this default behavior should still load fine, but in the edge case that there are prohibitively many columns to render (threshold will likely be ~1046), we may need to start with an all collapsed view for performance reasons.
+* Any nested columns (except for the top-level/parent column) will be a darker color to recede into the background, which can help users distinguish when quickly scanning the list.
+* The table header will be stick to the top at all times so users can always tell what the columns are as they scroll.
+
+
+
+![All collapsed view](../assets/047/all-collapsed-view.png)
+
+* When the user clicks the ‘Collapse all nested’ text button, the list will only show top-level or non-nested columns.
+* The up/down chevron that used to expand each column row to show full description and stats within the list area, will now change in functionality to expand/collapse each (one at a time, not all) level of the nested column. Therefore, only nested columns will have a chevron, and a column without any nesting under it will not have a chevron.
+
+
+
+![Each level drill down using chevron](../assets/047/each-level-drill-down-using-chevron.png)
+
+* Users can keep clicking on the chevron to expand each level of nesting if they want to  drill down the path without expanding all nesting at once.
+
+
+
+![Column details in right side panel](../assets/047/column-details-in-right-side-panel.png)
+
+* Clicking on a column name opens the right side panel to display all column details, including badges, description, and stats.
+* Only the columns with metadata will be displayed in interactive (Indigo 60) color, so that users can tell whether they will see useful information by clicking on its name.
+* For the new right side panel and layout, see details in the [Column details access and side panels RFC](./046-column-details-access-and-side-panels.md).
+* In the edge case so many nested levels indent the list too much, users can still side scroll or widen the browser. 
+
+
+
+### Nested object without named fields (e.g. Array and Map)
```

Comment:
I am not 100% sure why (and whether) we need to distinguish it at all? This might overcomplicate the UI without adding much value.

Reply:
If a column is a struct, and another column is an array of structs or even an array of arrays of structs (and so on), in the current state the UI would represent them the same way by only showing rows in the table for the named values in the struct. The thought here was by adding a short row for each level it wouldn't need to take up much screen space and you'd be able to quickly distinguish the different column types. In addition, this would allow the possibility of adding and viewing metadata such as badges and descriptions for the intermediate levels.

Reply:
It sounds like a distinction without a difference, but in the half-dozen design meetings we had to discuss how to represent nested structures we found this distinction was quite fundamental.